packages/react/src/components/table/table-footer.tsx

```diff
@@ -0,0 +1,71 @@
+import { ReactElement } from 'react';
+import styled, { css } from 'styled-components';
+import {
+    Header,
+    HeaderGroup,
+    ColumnDef as OriginalColumnDef,
+    Column,
+    flexRender,
+} from '@tanstack/react-table';
+
+type TextAlignOptions = 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
+
+type CustomColumnDef<TData extends object, TValue> = OriginalColumnDef<TData, TValue> & {
+    className?: string;
+    textAlign?: TextAlignOptions; // Define this type if not already defined
+    sticky?: boolean;
+    footerColSpan?: number;
+};
+
+interface CustomFooter<TData extends object, TValue> extends Header<TData, TValue> {
+    column: Column<TData, TValue> & {
+        columnDef: CustomColumnDef<TData, TValue>;
+    };
+    footerColSpan?: number;
+}
+
+const StyledFooter = styled.td<{ sticky: boolean }>`
+    background-color: ${({ theme }) => theme.greys.white};
+    ${({ sticky }) => sticky && css`
+        position: sticky;
+    `}
+`;
+
+function getFooter<TData extends object, TValue>(
+    footer: CustomFooter<TData, TValue>,
+    stickyFooter: boolean,
+): ReactElement | null {
+    if (footer.column.columnDef.footerColSpan === 0) {
+        // If the columns object has footerColSpan === 0, skip rendering the td
+        return null;
+    }
+    return (
+        <StyledFooter
+            key={footer.id}
+            colSpan={footer.column.columnDef.footerColSpan || footer.colSpan}
+            sticky={stickyFooter}
+        >
+            {footer.isPlaceholder
+                ? null
+                : flexRender(
+                    footer.column.columnDef.footer,
+                    footer.getContext(),
+                )}
+        </StyledFooter>
+    );
+}
+
+export interface FooterProps<T extends object> {
+    footerGroup: HeaderGroup<T>;
+    stickyFooter: boolean;
+}
+export const TableFooter = <T extends object>({
+    footerGroup,
+    stickyFooter,
+}:FooterProps<T>): ReactElement => { // eslint-disable-line arrow-body-style
```

Comment:
It seem like ESLint become crazy on this one, it's get conflicted indent between `eslint/indent` and `react/jsx-indent` (it seem because of the generic type, otherwise it usually work fine).

If I have more time, I will looks at a proper fix on the formatter configuration.